nav2_regulated_pure_pursuit_controller/src/regulated_pure_pursuit_controller.cpp

```diff
@@ -245,7 +268,14 @@ geometry_msgs::msg::TwistStamped RegulatedPurePursuitController::computeVelocity
     curvature = 2.0 * carrot_pose.pose.position.y / carrot_dist2;
   }
 
-  linear_vel = desired_linear_vel_;
+  // Setting the velocity direction
+  double sign = 1.0;
+
+  if(reverse_driving_){
```

Comment:
Why is this called `reverse_driving_`? What is this boolean meant for? If the path contains changes in velocity directions, we must follow it, this is a pure pursuit planner, it cannot deviate

Reply:
This is continuation of the first conversation. 

> What is this boolean meant for? 

The boolean here is to determine the position of the carrot, according to which we will be setting the linear velocity. As the position of the carrot gives the information related to the base_link frame. In case if a user decides to skip the reverse driving then, we never want to look at the sign of the carrot, since we already have a functionality that would provide as the rotation to the heading. In this scenario, the boolean will not be necessary in any case.

Reply:
Can't we just remove that parameter? If the path has a cusp in it, the robot has to reverse to follow it. Otherwise, the robot would be totally stuck. I suppose if it were a diff drive robot, it could follow the carrot into a cusp and then just rotate in place to follow... OK. I'm fine keeping the option to disable reversing, but lets rename this then so its more clear `allow_reversing` parameter?

Reply:
> I suppose if it were a diff drive robot, it could follow the carrot into a cusp and then just rotate in place to follow...

This indeed sounds to be a fancy and intelligent stuff from a controller. But, curious to know from a user perspective on what the benefit would be for them. 

> OK. I'm fine keeping the option to disable reversing, but lets rename this then so its more clear allow_reversing parameter? 

Agreed! That sounds right.

Reply:
OK. I'm fine keeping the option to disable reversing, but lets rename this then so its more clear `allow_reversing` parameter?

Agreed with the name!